src/library_webgl.js

```diff
@@ -486,17 +487,37 @@ var LibraryGL = {
       // code has been downloaded.
       if (Module['preinitializedWebGLContext']) {
         var ctx = Module['preinitializedWebGLContext'];
-#if USE_WEBGL2
-        webGLContextAttributes.majorVersion = (typeof WebGL2RenderingContext !== 'undefined' && ctx instanceof WebGL2RenderingContext) ? 2 : 1;
-#else
         webGLContextAttributes.majorVersion = 1;
+        webGLContextAttributes.minorVersion = 0;
+#if GL_MAX_FEATURE_LEVEL >= 20
+        if (typeof WebGL2RenderingContext !== 'undefined' && ctx instanceof WebGL2RenderingContext) {
+          webGLContextAttributes.majorVersion = 2;
+          webGLContextAttributes.minorVersion = 0;
+        }
+#endif
+#if GL_MAX_FEATURE_LEVEL >= 30
+        if (typeof WebGL2ComputeRenderingContext !== 'undefined' && ctx instanceof WebGL2ComputeRenderingContext) {
+          webGLContextAttributes.majorVersion = {{{ cDefine('EM_WEBGL_WEBGL2_COMPUTE_MAJOR_VERSION') }}};
+          webGLContextAttributes.minorVersion = {{{ cDefine('EM_WEBGL_WEBGL2_COMPUTE_MINOR_VERSION') }}};
+        }
 #endif
       } else {
 #endif
-
-      var ctx = 
-#if USE_WEBGL2
-        (webGLContextAttributes.majorVersion > 1) ? canvas.getContext("webgl2", webGLContextAttributes) :
+#if GL_MAX_FEATURE_LEVEL >= 20
+      var contextIdForFutureVersion =
+#if GL_MAX_FEATURE_LEVEL >= 30
+        "webgl2-compute" ||
+#endif
+        "webgl2";
+#endif
```

Comment:
This seems to generate `var contextIdForFutureVersion = "webgl2-compute" || "webgl2"` with certain flags. What's the point of that? It will always evaluate to `"webgl2-compute"`.

Reply:
Ideally I want something like this:

```
#if GL_MAX_FEATURE_LEVEL >= 20
    var contextIdForFutureVersion =
#if GL_MAX_FEATURE_LEVEL >= X
        "foo"
#elif GL_MAX_FEATURE_LEVEL >= Y
        "bar"
#elif GL_MAX_FEATURE_LEVEL >= Z
        "baz"
#else
        "webgl2";
#endif
#endif
```

But `#elif` is not supported by the preprocessor, and maybe I was falling asleep/in a bad mood at the time of writing, which results in the strange code here.